packages/swingset/src/get-nav-tree.ts

```diff
@@ -1,49 +1,86 @@
 
-  const componentEntitiesWithChildren = componentEntities.map((entity) => {
-    if (entity.isNested) return entity
+  const componentEntitiesWithChildren = componentEntities.map(
+    (componentEntity) => {
+      if (componentEntity.isNested) return componentEntity
 
-    entity.children = componentEntities.filter(
-      (childEntity) =>
-        childEntity.isNested &&
-        childEntity.componentPath === entity.componentPath
-    )
+      componentEntity.children = componentEntities.filter(
+        (childEntity) =>
+          childEntity.isNested &&
+          childEntity.componentPath === componentEntity.componentPath
+      )
 
-    return entity
-  })
+      return componentEntity
+    }
+  )
+
+  //TODO: Account for duplicate folder names [category]/[folder]
+  const categories = new Map<CategoryNode['title'], CategoryNode>()
 
   // bucket components into categories, nested documents are categorized under their component's path
   for (const entity of componentEntitiesWithChildren) {
     if (entity.isNested) continue
 
-    //TODO: Handle Default Category
-    const category = entity.navigationData?.category!
+    const componentNode: ComponentNode = {
+      __type: 'component',
+      title: entity.title,
+      slug: entity.slug,
+      componentPath: entity.componentPath,
+      children: entity.children,
+    }
+
+    const categoryTitle = entity.navigationData?.category || 'default'
 
+    const hasCategory = categories.has(categoryTitle)
 
+    //if category doesnt exist, create it
+    const storedCategory = hasCategory
+      ? categories.get(categoryTitle)!
+      : categories.set(categoryTitle, {
+          type: 'category',
+          title: categoryTitle,
+          children: [],
+        }) && categories.get(categoryTitle)!
 
-    result[category] ||= []
 
+    const folderTitle = entity.navigationData?.folder
+    const hasFolder = !!folderTitle
+    const folder = storedCategory.children.find(
+      (node) => node.title === folderTitle
+    )
 
-  
-    result[category].push({
-      title: entity.title,
-      slug: entity.slug,
-      componentPath: entity.componentPath,
-      children: entity.children,
-    })
+    //if node belongs in a folder, and folder doesnt exist, create folder with node
+    if (hasFolder && !!folder === false) {
+      storedCategory.children.push({
+        __type: 'folder',
+        title: folderTitle,
+        parentCategory: categoryTitle,
+        children: [componentNode],
+      })
+      continue
+    }
+
+    //if node belongs in a folder, and folder already exists, add node to folder
+    if (hasFolder && !!folder) {
+      folder.children ||= []
+      folder.children.push(entity)
+      continue
+    }
 
-    result[category].sort((a, b) => (a.slug > b.slug ? 1 : -1))
+    //if node doesnt belong in folder, add node
+    storedCategory.children.push(componentNode)
   }
 
-  return result
+  const tree = Object.fromEntries(categories)
+  return tree
```

Comment:
We want to return `NavigationTree` here, right? That's an array of `CategoryNode`, so we don't want an object here.

This might work (builds an array from the `Map`'s values):
```ts
Array.from(categories.values())
```